src/TimeModeling/Types/judiWavefield.jl

```diff
@@ -43,135 +33,73 @@ Constructors
 
-function vcat(ai::Vararg{judiWavefield{avDT}, N}) where {avDT, N}
-	N == 1 && (return ai[1])
-	N > 2 && (return vcat(ai[1], vcat(ai[2:end]...)))
-	a, b = ai
-	m = a.m + b.m
-	n = 1
-	nsrc = a.info.nsrc + b.info.nsrc
-	data = Array{Array{avDT, 3}, 1}(undef, nsrc)
-	nt = Array{Integer}(undef, nsrc)
-	for j=1:a.info.nsrc
-		data[j] = a.data[j]
-		nt[j] = a.info.nt[j]
-	end
-	for j=a.info.nsrc+1:nsrc
-		data[j] = b.data[j-a.info.nsrc]
-		nt[j] = b.info.nt[j-a.info.nsrc]
-	end
-	info = Info(a.info.n, nsrc, nt)
-	return judiWavefield{avDT}("judiWavefield",info.n * sum(info.nt), 1, info, a.dt ,data)
-end
-
 function push!(a::judiWavefield{T}, b::judiWavefield{T}) where T
-	append!(a.data, b.data)
-	a.m += b.m
-	a.info.nsrc += b.info.nsrc
+    append!(a.data, b.data)
+    append!(a.dt, b.dt)
+    a.nsrc += b.nsrc
 end
 
 # DFT operator for wavefields, acts along time dimension
-function fft_wavefield(x_in,mode)
-	nsrc = x_in.info.nsrc
-	nt = size(x_in.data[1], 1)
-	if mode==1
-		x = similar(x_in, Complex{Float32})
-		for i=1:nsrc
-			x.data[i] = fft(x_in.data[i], 1)/sqrt(nt)
-		end
-	elseif mode==-1
-		x = similar(x_in, Float32)
-		for i=1:nsrc
-			x.data[i] = real(ifft(x_in.data[i], 1)) * sqrt(nt)
-		end
-	end
-	return x
-end
-
-# Sampling mask to extract wavefields from full vector
-subsample(u::judiWavefield,srcnum) = judiWavefield(u.info,u.data[srcnum];vDT=eltype(u))
-similar(u::judiWavefield) = 0f0 * u
-similar(u::judiWavefield, vDT::DataType) = judiWavefield{vDT}("judiWavefield",u.m, u.n, u.info, u.dt, Array{vDT}.(0f0.*u.data))
-
-# norm
-function norm(a::judiWavefield{avDT}, p::Real=2) where avDT
-    if p == Inf
-        return max([maximum(abs.(a.data[i])) for i=1:a.info.nsrc]...)
-    end
-    x = 0.f0
-    for j=1:a.info.nsrc
-        x += Float32(a.dt) * sum(abs.(vec(a.data[j])).^p)
+function fft(x_in::judiWavefield{T}) where T
+    x = similar(x_in, Complex{Float32})
+    for i=1:x_in.nsrc
+        x.data[i] = fft(x_in.data[i], 1)/sqrt(size(x_in.data[i], 1))
     end
-    return x^(1.f0/p)
+    return x
 end
 
-# inner product
-function dot(a::judiWavefield{avDT}, b::judiWavefield{bvDT}) where {avDT, bvDT}
-	# Dot product for data containers
-	size(a) == size(b) || throw(judiWavefieldException("dimension mismatch"))
-	dotprod = 0f0
-	for j=1:a.info.nsrc
-		dotprod += Float32(a.dt) * dot(vec(a.data[j]),vec(b.data[j]))
-	end
-	return dotprod
-end
-
-# abs
-function abs(a::judiWavefield{avDT}) where avDT
-	b = deepcopy(a)
-	for j=1:a.info.nsrc
-		b.data[j] = abs.(a.data[j])
-	end
-	return b
-end
-
-function isapprox(x::judiWavefield, y::judiWavefield; rtol::Real=sqrt(eps()), atol::Real=0)
-    x.info.nsrc == y.info.nsrc || throw(judiVectorException("Incompatible number of sources"))
-    isapprox(x.data, y.data; rtol=rtol, atol=atol)
+function ifft(x_in::judiWavefield{T}) where T
+    x = similar(x_in, Float32)
+    for i=1:x_in.nsrc
+        x.data[i] = real(ifft(x_in.data[i], 1)) * sqrt(size(x_in.data[i], 1))
```

Comment:
Julia's fft/ifft doesn't need any scaling
```julia
julia> using FFTW

julia> x = randn(Float32, 10)
10-element Vector{Float32}:
 -0.36564797
 -0.49766394
 -0.5977085
  1.5845047
  0.76346606
  0.059032917
 -1.9635432
  2.2681353
 -0.19259544
 -1.2812192

julia> using LinearAlgebra

julia> norm(ifft(fft(x))-x)
3.8701805f-7
```
Why do we need it here? (Although I see you add a test below)

Reply:
ALways been there, you free to change it if you want